### PR TITLE
Misc. fixes to Cigarette stuff

### LINKED ISSUE #4493: There are no donuts left in the box

### PR DESCRIPTION
**Pull request by ghost**
- Re-add delay for regular lighters. Intended to avoid spamming
- Fix the position of some update_brightness() instructions, leading to odd strings
- Fix a few typecasts, add a visible_message or two

Changelog included

Fixes #4493

### CODE REVIEW

**Comment by Exxion** [comment]
#4493
Add icon_type to matchboxes

**Comment by clusterfack** [comment]
Did you fix the bug where you can light things with burnt out matches?

**Comment by dylanstrategie** [comment]
I'll drop a fix to icon_type

I'll look at the problem with burnt matches, I have a good reason to believe I know what is wrong

**Comment by clusterfack** [comment]
Probably checking if(lit) after you made them be burnt with lit = -1

File: code/game/objects/items/weapons/storage/fancy.dm [hunk 1]
```diff
@@ -184,6 +185,16 @@
 /obj/item/weapon/storage/fancy/matchbox/empty
 	empty = 1
 
+//Needs to be modified, because plural is -es instead of -s
+/obj/item/weapon/storage/fancy/matchbox/examine(mob/user)
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/storage/fancy.dm]: This is going to make a duplicate message appear.
Don't do it. Just make a way for it to display a plural instead.

Comment (dylanstrategie) [inline code/game/objects/items/weapons/storage/fancy.dm]: I'll look into it

**Pull request merged**